### PR TITLE
feat(category): allow nesting of categories

### PR DESCRIPTION
**Pull request by anehx**
[Frontend PR](https://github.com/projectcaluma/ember-alexandria/pull/497)

### CODE REVIEW

File: alexandria/core/serializers.py [hunk 1]
```diff
@@ -100,9 +100,25 @@ def create(self, validated_data):
 
 
 class CategorySerializer(SlugModelSerializer):
+    parent = serializers.ResourceRelatedField(required=False, read_only=True)
+    children = serializers.ResourceRelatedField(
```
Comment (Yelinz) [inline alexandria/core/serializers.py:104]: There is no backend limitation on children depth?

Comment (anehx) [reply alexandria/core/serializers.py:104]: No, currently not

File: alexandria/core/tests/snapshots/snap_test_api.py [hunk 2]
```diff
@@ -669,11 +706,14 @@
...
+        'SELECT "alexandria_core_category"."created_at", "alexandria_core_category"."created_by_user", "alexandria_core_category"."created_by_group", "alexandria_core_category"."modified_at", "alexandria_core_category"."modified_by_user", "alexandria_core_category"."modified_by_group", "alexandria_core_category"."metainfo", "alexandria_core_category"."slug", "alexandria_core_category"."name", "alexandria_core_category"."description", "alexandria_core_category"."color", "alexandria_core_category"."parent_id" FROM "alexandria_core_category" WHERE "alexandria_core_category"."parent_id" = \'father-especially\'',
```
Comment (winged) [inline alexandria/core/tests/snapshots/snap_test_api.py:714]: Maybe a missing `select_related`?

**Review by winged — APPROVED** [review]
LGTM - Just a minor thing I noticed with the snapshots that may be worth looking at

**Pull request merged**